src/instrumenter/instrument.ts

```diff
@@ -1139,36 +1139,54 @@ export function instrumentStatement(
     stmt: Statement
 ): void {
     const factory = ctx.factory;
+    const singlePointAnnots: AnnotationMetaData[] = [];
+    const ifSucceedsAnnots: AnnotationMetaData[] = [];
 
     for (const annot of allAnnotations) {
-        assert(
+        if (
             annot.type === AnnotationType.Assert ||
-                annot.type === AnnotationType.Try ||
-                annot.type === AnnotationType.Require,
-            `Unexpected annotaiton on statement ${annot.original}`
-        );
+            annot.type === AnnotationType.Try ||
+            annot.type === AnnotationType.Require
+        ) {
+            singlePointAnnots.push(annot);
+        } else if (annot.type == AnnotationType.IfSucceeds) {
+            ifSucceedsAnnots.push(annot);
+        } else {
+            assert(false, `Unexpected annotaiton on statement ${annot.original}`);
+        }
     }
 
     // Make sure stmt is contained in a block. (converts cases like `while () i++` to `while () { i++}`
     ensureStmtInBlock(stmt, factory);
 
     const container = stmt.parent as Block;
-    const assertionBlock = gte(ctx.compilerVersion, "0.8.0")
+    const beforeStmtBlock = gte(ctx.compilerVersion, "0.8.0")
         ? factory.makeUncheckedBlock([])
         : factory.makeBlock([]);
 
     // Add a new block before the target statement where we will transpile the assertions
-    container.insertBefore(assertionBlock, stmt);
+    container.insertBefore(beforeStmtBlock, stmt);
 
     const fun = stmt.getClosestParentByType(FunctionDefinition);
 
     assert(fun !== undefined, "Unexpected orphan stmt", stmt);
 
-    const transCtx = ctx.transCtxMap.get(fun, InstrumentationSiteType.Assert);
+    const transCtx = ctx.transCtxMap.get(fun, InstrumentationSiteType.Custom);
 
-    transCtx.resetMarkser([assertionBlock, "end"], false);
+    transCtx.resetMarker([beforeStmtBlock, "end"], false);
 
-    insertAnnotations(allAnnotations as PropertyMetaData[], transCtx);
+    insertAnnotations(singlePointAnnots as PropertyMetaData[], transCtx);
+
+    if (ifSucceedsAnnots.length > 0) {
+        const afterStmtBlock = gte(ctx.compilerVersion, "0.8.0")
```

Comment:
Code duplication with line 1163 - could be a helper on ScribbleFactory?

Reply:
Fixed in [a482f6b](https://github.com/ConsenSys/scribble/pull/154/commits/a482f6b39338dedcd43bdfe3f23939aed396ad48).